src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -3746,6 +3756,12 @@ public boolean isKeyDenylisted(String keyspace, String table, String partitionKe
         return !partitionDenylist.isKeyPermitted(keyspace, table, bytes);
     }
 
+    private void checkPartitionDenylistEnabled() {
+        if (!getPartitionDenylistEnabled()) {
+            throw new UnsupportedOperationException("Denylisting partitions is disabled");
```

Comment:
I am not sure this is the most appropriate one. This looks like the code is not there yet. `RuntimeException` is probably just enough.